pandas/core/groupby/groupby.py

```diff
@@ -1623,11 +1625,17 @@ def ohlc(self) -> DataFrame:
 
     @doc(DataFrame.describe)
     def describe(self, **kwargs):
-        with _group_selection_context(self):
-            result = self.apply(lambda x: x.describe(**kwargs))
-            if self.axis == 1:
-                return result.T
-            return result.unstack()
+        result = self.apply(lambda x: x.describe(**kwargs))
+        if self.axis == 1:
+            return result.T
```

Comment:
Was `axis=1` not affected by the same issue?

Reply:
AFAICT, no. 

This issue only came up when grouping by a column of non-nuisance values. `DataFrame.groupby` with `axis=1` doesn't apply to the rows the same way `axis=0` applies to the columns. If `axis=1` then `by` needs to be a column level-name or a mapping. 

See examples below. These all produce the same output on master and on PR branch. 

```
import numpy as np
import pandas as pd

df = pd.DataFrame(
    {
        "a": [99, 99, 99, 88, 88, 88],
        "b": [1, 2, 3, 4, 5, 6],
        "c": [10, 20, 30, 40, 50, 60],
    }
)

# axis=0 can group on column of values
df.groupby('a', axis=0).describe()

# axis=1 cannot group on row of values. Raises KeyError
df_T = df.T
df_T.groupby('a', axis=1)

# axis=1 works when 'by' is an index level name
df_T = df.set_index('a').T
df_T.groupby('a', axis=1).describe()

# axis=1 works when 'by' is a mapping
df_T = df.T
df_T.groupby(np.array([0,0,0,1,1,1]), axis=1).describe()

```
So `.describe` with `axis=1` does not seem to be affected by the original bug or by the changes. 

There are existing tests for `GroupBy.describe` with 'axis=1` and none of them were affected by the change. 

https://github.com/pandas-dev/pandas/blob/master/pandas/tests/groupby/test_function.py#L941

Reply:
Following this update the `.describe` method is no longer affected by this PR. 

https://github.com/pandas-dev/pandas/pull/35314#issuecomment-664193224